api/v1/client.go

```diff
@@ -1123,3 +1294,200 @@ func (c *storageClient) ProposalVotes(ctx context.Context, r *http.Request) (*Pr
 
 	return &vs, nil
 }
+
+// Validator returns a single validator
+func (c *storageClient) Validator(ctx context.Context, r *http.Request) (*Validator, error) {
+	chainID, ok := ctx.Value(ChainIDContextKey).(string)
+	if !ok {
+		return nil, common.ErrBadChainID
+	}
+
+	var epoch Epoch
+	if err := c.db.QueryRow(
+		ctx,
+		fmt.Sprintf(`
+		SELECT id, start_height, end_height
+			FROM %s.epochs
+			ORDER BY id DESC
+			LIMIT 1`,
+			chainID),
+	).Scan(&epoch.ID, &epoch.StartHeight, &epoch.EndHeight); err != nil {
+		c.logger.Info("row scan failed",
+			"request_id", ctx.Value(RequestIDContextKey),
+			"err", err.Error(),
+		)
+		return nil, common.ErrStorageError
+	}
+
+	row := c.db.QueryRow(
+		ctx,
+		fmt.Sprintf(`
+			SELECT
+				%s.entities.address AS entity_address,
+				%s.nodes.id AS node_address,
+				%s.accounts.escrow_balance_active AS escrow,
+				%s.commissions.schedule AS commissions_schedule,
+				case when exists(select null from %s.nodes where %s.entities.id = %s.nodes.entity_id AND voting_power > 0) then true else false end as active,
+				case when exists(select null from %s.nodes where %s.entities.id = %s.nodes.entity_id AND %s.nodes.roles like 'validator') then true else false end as status,
+				meta
+			FROM %s.entities
+			JOIN %s.accounts ON %s.entities.address = %s.accounts.address
+			JOIN %s.metadata ON %s.entities.id = %s.metadata.id
+			JOIN %s.commissions ON %s.entities.address = %s.commissions.address
+			JOIN %s.nodes ON %s.entities.id = %s.nodes.entity_id
```

Comment:
This is too many joins. I think the `metadata` one will go away if you take my below comment to roll that into the entities table. Overall I think the query can look like:

```
SELECT id, entity_id FROM nodes WHERE roles LIKE '%validator%';
SELECT address, metadata, voting_power > 0 AS ACTIVE FROM entities WHERE id = <entity_id_from_prev_query>;
SELECT schedule FROM commissions WHERE address = <address_from_prev_query>;
```
This will be more efficient than joining everything.

Reply:
And something similar below for `ValidatorsList`.

Reply:
>  I think the metadata one will go away if you take my below comment to roll that into the entities table.

Agree.

> This will be more efficient than joining everything.

Preference for three queries? I don't have numbers, but thought default `JOIN` would be faster.